internal/modulecontext/from_secrets.go

```diff
@@ -0,0 +1,55 @@
+package modulecontext
+
+import (
+	"context"
+	"fmt"
+	"strings"
+)
+
+// DatabasesFromSecrets finds DSNs in environment variables and creates a map of databases.
+//
+// Environment variables should be in the format FTL_POSTGRES_DSN__<MODULENAME>_<DBNAME>
+func DatabasesFromSecrets(ctx context.Context, module string, secrets map[string][]byte) (map[string]Database, error) {
+	databases := map[string]Database{}
+	for sName, maybeDSN := range secrets {
+		if !strings.HasPrefix(sName, "FTL_DSN_") {
+			continue
+		}
+		// FTL_DSN_<MODULE>_<DBNAME>
+		parts := strings.Split(sName, "_")
+		if len(parts) != 4 {
+			return nil, fmt.Errorf("invalid DSN secret key %q should have format FTL_DSN_MODULE_DBNAME", sName)
```

Comment:
Use `FTL_DSN_<MODULE>_<DBNAME>` in the error so it's clear what parts are variable.

Reply:
good idea - done